src/data/actions/assets.js

```diff
@@ -54,9 +53,39 @@ export const deleteAsset = (assetId, courseDetails) =>
         if (response.ok) {
           dispatch(deleteAssetSuccess(assetId, response));
         } else {
-          dispatch(assetXHRFailure(response));
+          dispatch(assetDeleteFailure(response));
+        }
+      });
+
+export const togglingLockAsset = (asset, response) => ({
+  type: assetActions.TOGGLING_LOCK_ASSET_SUCCESS,
+  asset,
+  response,
+});
+
+export const toggleLockAssetSuccess = asset => ({
+  type: assetActions.TOGGLE_LOCK_ASSET_SUCCESS,
+  asset,
+});
+
+export const toggleLockAssetFailure = (asset, response) => ({
+  type: assetActions.TOGGLING_LOCK_ASSET_FAILURE,
+  asset,
+  response,
+});
+
+export const toggleLockAsset = (asset, courseDetails) =>
```

Comment:
I think the discussion at the top of this page would be useful. https://redux.js.org/docs/advanced/AsyncActions.html I think we basically just need to come to a decision about how we want to do it and document it somewhere so that we do it consistently.